cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sources/KVTableSource.java

```diff
@@ -35,12 +41,24 @@
     Schema.Field.of("value", Schema.of(Schema.Type.BYTES))
   );
 
-  @Override
-  public void configure(StageConfigurer configurer) {
-    configurer.setName(getClass().getSimpleName());
-    configurer.setDescription("CDAP KeyValue Table Dataset Batch Source. Outputs records with a 'key' field " +
-      "and a 'value' field. Both fields are of type bytes.");
-    configurer.addProperty(new Property(NAME, "Dataset Name", true));
+  private static final String NAME_DESC = "Dataset Name";
+
+  /**
+   * Config class for KVTableSource
+   */
+  public static class KVTableConfig extends PluginConfig {
+    @Description(NAME_DESC)
+    String name;
+
+    public KVTableConfig(String name) {
+      this.name = name;
+    }
+  }
+
+  private final KVTableConfig kvTableConfig;
```

Comment:
In the case of KVTable no where. As we don't use the properties. But I think it might be  needed for the  testing framework. 
@albertshau ?